src/configuration.rs

```diff
@@ -0,0 +1,376 @@
+
+
+use ed25519_dalek::*;
+//use rand::{Rng, RngCore, CryptoRng};
+use rand;
+use std::time::{Duration, SystemTime};
+use crypto::{
+    //aead::AeadEncryptor, chacha20poly1305::ChaCha20Poly1305,
+    aes, blockmodes,
+    buffer::{ ReadBuffer, WriteBuffer, BufferResult, RefReadBuffer, RefWriteBuffer },
+    sha2, digest::Digest,
```

Comment:
Could we reuse `sha2` crate that `ed25519-dalek` already depends on? https://github.com/dalek-cryptography/ed25519-dalek/blob/master/Cargo.toml#L30

Reply:
Ah; hadn't noticed that it had one.

Reply:
Ed25519 signature scheme actually uses SHA-512 internally (it can theoretically be swapped for any other 512-bit cryptographic hash).